xmtp/src/storage/encrypted_store/mod.rs

```diff
@@ -161,35 +163,68 @@ impl EncryptedMessageStore {
         Ok(account_list.pop())
     }
 
-    pub fn get_session_with_conn(
+    pub fn get_latest_session_for_installation(
         &self,
         installation_id: &str,
         conn: &mut DbConnection,
     ) -> Result<Option<StoredSession>, StorageError> {
         use self::schema::sessions::dsl::*;
 
-        let mut session_list = sessions
+        sessions
             .filter(peer_installation_id.eq(installation_id))
-            .order(created_at.desc())
-            .load::<StoredSession>(conn)
-            .map_err(|e| StorageError::Unknown(e.to_string()))?;
+            .order(updated_at.desc())
+            .first(conn)
+            .optional()
+            .map_err(|e| StorageError::Unknown(e.to_string()))
+    }
 
-        warn_length(&session_list, "StoredSession", 1);
-        Ok(session_list.pop())
+    pub fn get_latest_sessions_for_installation(
+        &self,
+        installation_id: &str,
+        conn: &mut DbConnection,
+    ) -> Result<Vec<StoredSession>, StorageError> {
+        use self::schema::sessions::dsl::*;
+
+        sessions
+            .filter(peer_installation_id.eq(installation_id))
+            .order(updated_at.desc())
+            .get_results(conn)
+            .map_err(|e| StorageError::Unknown(e.to_string()))
     }
 
-    pub fn get_sessions(
+    pub fn get_latest_sessions(
```

Comment:
Not needed for this PR, but would it be worth adding a unit test on this one, seeing as the SQL is complicated?